synapse/handlers/cas_handler.py

```diff
@@ -164,41 +206,31 @@ def _parse_cas_response(
             registered_user_id, request, client_redirect_url
         )
 
-    def handle_redirect_request(self, client_redirect_url: bytes) -> bytes:
+    async def handle_ui_auth_response(
+        self, request: SynapseRequest, ticket: str, session_id: str
+    ) -> None:
         """
-        Generates a URL to the CAS server where the client should be redirected.
+        Validates a CAS ticket sent by the client for user interactive authentication.
 
-        Args:
-            client_redirect_url: The final URL the client should go to after the
-                user has negotiated SSO.
+        If successful, this completes the SSO step of UI auth and returns a
+        an HTML page to the client.
 
-        Returns:
-            The URL to redirect to.
-        """
-        args = urllib.parse.urlencode(
-            {"service": self._build_service_param(client_redirect_url)}
-        )
+        Args:
+            request: the incoming request from the browser.
 
-        return ("%s/login?%s" % (self._cas_server_url, args)).encode("ascii")
+            ticket: The CAS ticket provided by the client.
 
-    async def handle_ticket_request(
-        self, request: SynapseRequest, client_redirect_url: str, ticket: str
-    ) -> None:
+            session_id: The UI Auth session ID.
         """
-        Validates a CAS ticket sent by the client for login/registration.
+        client_redirect_url = ""
+        user, _ = await self._validate_ticket(
```

Comment:
I don't think this will work correctly. AIUI we're supposed to submit the same `service` param for `https://cas_server/proxyValidate` as was given to `https://cas_server/login`. It looks like we're setting different query-params for the two calls?